addons/mail/static/src/core/common/composer.js

```diff
@@ -209,17 +210,49 @@ export class Composer extends Component {
     }
 
     get CANCEL_OR_SAVE_EDIT_TEXT() {
-        return markup(
-            _t(
-                "<samp>%(cancel_keybind)s</samp><i> to <a href='#' data-type='%(cancel_type)s'>cancel</a></i>, <samp>%(save_keybind)s</samp><i> to <a href='#' data-type='%(save_type)s'>save</a></i>",
-                {
-                    cancel_keybind: _t("Escape"),
-                    cancel_type: EDIT_CLICK_TYPE.CANCEL,
-                    save_keybind: this.sendKeybind,
-                    save_type: EDIT_CLICK_TYPE.SAVE,
-                }
-            )
-        );
+        if (this.ui.isSmall) {
+            return markup(
+                sprintf(
+                    escape(
+                        _t(
+                            "%(open_button)s%(icon)s%(open_em)sDiscard editing%(close_em)s%(close_button)s"
+                        )
+                    ),
+                    {
+                        open_button: `<button class='btn px-1 py-0' data-type="${escape(
+                            EDIT_CLICK_TYPE.CANCEL
+                        )}">`,
+                        close_button: "</button>",
```

Comment:
This is caused by reliance on `ev.target` in `onClickCancelOrSaveEditText`, and we receive the `<em>` instead of `<button>` in practice... I'll fix it.

note that the test passes because the click happens on button, so correct `ev.target`.

Reply:
In the end I put the `data-type` on all children. This is the easiest way to fix it immediately without finding another trick for handling all such clicks.

This is not easy to find an alternative: the result is `t-out` so this is a string put in template. We can't easily add `addEventListener` on those `<button/>`/`<a/>` because they are made by OWL, hence why we rely on `t-on-click` by OWL.

Probably worth changing that in master eventually to ease maintaining this feature better. As a fix, I think this is fine the currently implementation.